spi_nand_flash/src/dhara_glue.c

```diff
@@ -261,10 +279,17 @@ int dhara_nand_copy(const struct dhara_nand *n, dhara_page_t src, dhara_page_t d
 
     ESP_GOTO_ON_ERROR(read_page_and_wait(dev, src, &status), fail, TAG, "");
 
-    if (is_ecc_error(status)) {
-        ESP_LOGD(TAG, "copy, ecc error");
+    ecc_status_t bits_corrected_status = get_ecc_status(status, dev->ecc_data.ecc_status_reg_len_in_bits);
+    dev->ecc_data.ecc_corrected_bits_status = bits_corrected_status;
+    if (bits_corrected_status) {
         dhara_set_error(err, DHARA_E_ECC);
-        return -1;
+        if (bits_corrected_status == STAT_ECC_MAX) {
+            ESP_LOGE(TAG, "%s: Error while initializing value of ecc_status_reg_len_in_bits", __func__);
+            return -1;
+        } else if (bits_corrected_status == STAT_ECC_NOT_CORRECTED) {
+            ESP_LOGD(TAG, "copy, ecc error");
+            return -1;
+        }
```

Comment:
Similar question about setting error status here.

BTW, it seems that simply returning -1 without calling dhara_set_error doesn't do anything because dhara doesn't check the return value:

https://github.com/dlbeer/dhara/blob/1b166e41b74b4a62ee6001ba5fab7a8805e80ea2/dhara/journal.c#L856-L861